File: others/scala/src/main/scala/com/example/web/ExceptionHandlers.scala
package com.example.web

import javax.persistence.EntityNotFoundException
import org.springframework.http.{HttpStatus, ResponseEntity}
import org.springframework.web.bind.annotation.{ExceptionHandler, RestControllerAdvice}

@RestControllerAdvice
class ExceptionHandlers {

  @ExceptionHandler(Array(classOf[EntityNotFoundException]))
  def handleResponseStatusException(): ResponseEntity[AnyRef] = {
    new ResponseEntity(HttpStatus.NOT_FOUND)
  }
}

File: rdb/spring-jdbc/src/main/java/com/example/controller/ExceptionHandlers.java
package com.example.controller;

import java.util.NoSuchElementException;

import org.springframework.dao.EmptyResultDataAccessException;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class ExceptionHandlers {
    @ExceptionHandler({ EmptyResultDataAccessException.class, NoSuchElementException.class })
    public ResponseEntity<Object> handleNotFoundException(RuntimeException e) {
        return ResponseEntity.notFound().build();
    }
}
